internal/controllers/insights/apicast.go

```diff
@@ -0,0 +1,97 @@
+// Copyright The Cryostat Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package insights
+
+import (
+	"bytes"
+	"text/template"
+)
+
+type apiCastConfigParams struct {
+	FrontendDomains       string
+	BackendInsightsDomain string
+	HeaderValue           string
+	ProxyDomain           string
+}
+
+var apiCastConfigTemplate = template.Must(template.New("").Parse(`{
+  "services": [
+    {
+      "id": "1",
+      "backend_version": "1",
+      "proxy": {
+        "hosts": [{{ .FrontendDomains }}],
+        "api_backend": "https://{{ .BackendInsightsDomain }}:443/",
+        "backend": { "endpoint": "http://127.0.0.1:8081", "host": "backend" },
+        "policy_chain": [
+          {
+            "name": "default_credentials",
+            "version": "builtin",
+            "configuration": {
+              "auth_type": "user_key",
+              "user_key": "dummy_key"
+            }
+          },
+          {{- if .ProxyDomain }}
+          {
+            "name": "apicast.policy.http_proxy",
+            "configuration": {
+              "https_proxy": "http://{{ .ProxyDomain }}/",
+              "http_proxy": "http://{{ .ProxyDomain }}/"
+            }
+          },
+          {{- end }}
+          {
+            "name": "headers",
+            "version": "builtin",
+            "configuration": {
+              "request": [
+                {
+                  "op": "set",
+                  "header": "Authorization",
+                  "value_type": "plain",
+                  "value": "{{ .HeaderValue }}"
```

Comment:
This should be:

```json
"value": "Bearer {{ .HeaderValue }}"
```

or else the `Bearer` part should be added to the `HeaderValue` template variable elsewhere.

![image](https://github.com/cryostatio/cryostat-operator/assets/3787464/7db5a45b-37a2-4028-ae2c-f2c10371f9d9)

Reply:
Oh, good catch! I've fixed this.